File: oracle/fsharp/read/oracle_read.fs
// -------------------------------------------------------------------
//
//	oracle_read.fs
//
//					May/23/2011
//
// -------------------------------------------------------------------
#light "on"
open System
open System.Data
open System.Data.OracleClient

// -------------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =
  printfn "*** 開始 ***"

//  let str_connect = "server=cdbd025\SQLEXPRESS;uid=sa;pwd=hello9;database=city;"
  let str_connect = "User Id=scott;Password=tiger;Data Source=spn109:1521/xe"

  let conn = new OracleConnection (str_connect)
  conn.Open()

  let cmd = new OracleCommand( "SELECT * FROM cities", conn )
  let reader = cmd.ExecuteReader()

  while reader.Read()
    do System.Console.Write (reader.GetString 0 )
       System.Console.Write ("\t")
       System.Console.Write (reader.GetString 1 )
       System.Console.Write ("\t")
       System.Console.Write (reader.GetInt32 2 )
       System.Console.Write ("\t")
       System.Console.WriteLine (reader.GetDateTime 3 )
       done

  reader.Close()

  printfn "*** 終了 ***"
  0
// -------------------------------------------------------------------

File: oracle/csharp/read/oracle_read.cs
// ----------------------------------------------------------------
/*
	read/oracle_read.cs

					Aug/17/2015

export ORACLE_HOME=/usr/lib/oracle/xe/app/oracle/product/10.2.0/client
export PATH=$ORACLE_HOME/bin:$PATH
export LD_LIBRARY_PATH=$ORACLE_HOME/lib:$LD_LIBRARY_PATH
export NLS_LANG=japanese_japan.UTF8
export ORACLE_SID=XE

*/
// ----------------------------------------------------------------
using System;
using System.Data;
using System.Data.OracleClient;

// ----------------------------------------------------------------
public static class oracle_read
{

// ----------------------------------------------------------------
public static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	string str_connect =
		"User Id=scott;Password=tiger;Data Source=host_oracle:1521/xe";

	string str_com = "select * from cities";

	OracleConnection connection = new OracleConnection (str_connect);

	OracleDataAdapter da_adapter = new OracleDataAdapter
			(str_com,str_connect);


	DataTable dtable = new DataTable ();

	connection.Open ();
	da_adapter.Fill (dtable);
	connection.Close ();

	table_manipulate.display_proc (dtable);

	Console.WriteLine ("*** 終了 ***");
}
// ----------------------------------------------------------------
}
// ----------------------------------------------------------------
